lib/cartopy/tests/mpl/test_feature_artist.py

```diff
@@ -71,73 +61,67 @@ def cached_paths(geom, target_projection):
     return geom_cache.get(target_projection, None)
 
 
-@mock.patch('matplotlib.collections.PathCollection')
-def test_feature_artist_draw(path_collection_cls, feature):
-    geoms = list(feature.geometries())
+@pytest.mark.mpl_image_compare(filename='feature_artist.png')
+def test_feature_artist_draw(feature):
+    fig, ax = robinson_map()
+    ax.add_feature(feature, facecolor='blue')
 
-    fa = FeatureArtist(feature, facecolor='red')
-    prj_crs = ccrs.Robinson()
-    fa.axes = mocked_axes(extent=[-10, 10, -10, 10], projection=prj_crs)
-    fa.draw(mock.sentinel.renderer)
+    return fig
+
+
+@pytest.mark.mpl_image_compare(filename='feature_artist.png')
+def test_feature_artist_draw_facecolor_list(feature):
+    fig, ax = robinson_map()
+    ax.add_feature(feature, facecolor=['red', 'green', 'blue'])
+
+    return fig
 
-    transform = prj_crs._as_mpl_transform(fa.axes)
-    expected_paths = ([cached_paths(geoms[0], prj_crs),
-                       cached_paths(geoms[1], prj_crs)], )
-    expected_style = {'facecolor': 'red'}
 
-    args, kwargs = path_collection_cls.call_args_list[0]
-    assert transform == kwargs.pop('transform', None)
-    assert kwargs == expected_style
-    assert args == expected_paths
+@pytest.mark.mpl_image_compare(filename='feature_artist.png')
+def test_feature_artist_draw_cmap(feature):
+    fig, ax = robinson_map()
 
-    path_collection_cls().set_clip_path.assert_called_once_with(fa.axes.patch)
-    path_collection_cls().set_figure.assert_called_once_with(fa.axes.figure)
-    path_collection_cls().draw(mock.sentinel.renderer)
+    cmap = mcolors.ListedColormap(['red', 'gray', 'blue'])
+    ax.add_feature(feature, cmap=cmap, array=[0, 0, 1])
 
+    return fig
 
-@mock.patch('matplotlib.collections.PathCollection')
-def test_feature_artist_draw_styler(path_collection_cls, feature):
+
+@pytest.mark.mpl_image_compare(filename='feature_artist.png')
+def test_feature_artist_draw_styled_feature(feature):
     geoms = list(feature.geometries())
-    style1 = {'facecolor': 'blue', 'edgecolor': 'white'}
-    style2 = {'color': 'black', 'linewidth': 1}
-    style2_finalized = style.finalize(style.merge(style2))
+    styled_feature = ShapelyFeature(geoms, crs=ccrs.PlateCarree(), facecolor='blue')
 
-    def styler(geom):
-        if geom == geoms[0]:
-            return style1
-        else:
-            return style2
+    fig, ax = robinson_map()
+    ax.add_feature(styled_feature)
+
+    return fig
 
-    fa = FeatureArtist(feature, styler=styler, linewidth=2)
-    extent = [-10, 10, -10, 10]
-    prj_crs = ccrs.Robinson()
-    fa.axes = mocked_axes(extent, projection=prj_crs)
-    fa.draw(mock.sentinel.renderer)
 
-    transform = prj_crs._as_mpl_transform(fa.axes)
+@pytest.mark.mpl_image_compare(filename='feature_artist.png')
+def test_feature_artist_draw_styler(feature):
+    geoms = list(feature.geometries())
+
+    def styler(geom):
+        if geom == geoms[1]:
+            return {}
```

Comment:
Might be good to test setting an explicit other color here?
```suggestion
            return {'facecolor': 'red'}
```